builder/vagrant/driver_2_2.go

```diff
@@ -270,7 +270,7 @@ func (d *Vagrant_2_2_Driver) vagrantCmd(args ...string) (string, string, error)
 		log.Printf("[vagrant driver] stdout: %s", line)
 		stdoutString += line + "\n"
 	}
-	_ = cmd.Wait()
+	err = cmd.Wait()
 
 	if _, ok := err.(*exec.ExitError); ok {
 		err = fmt.Errorf("Vagrant error: %s", stderrString)
```

Comment:
Does the Wait command always error when the Start command errors? If not we need to check the start command for ExitErrors too?

Reply:
`Start` errors if the command fails to be started, and we already check/exit if it does. The type of error doesn't matter here, and I would think this won't be an `ExitError` in any case, since the process won't even be created.
The problem here is if the command starts, and errors for any reason in the meantime, we didn't capture it, so even when `vagrant` returned non-zero, it would never report it.
TBF I'm not certain that checking the error type is relevant at all after `Wait()`, but I haven't dug into the complete history of the file, so it's hard to say, maybe there was a rationale for that type check. In the end for this commit I opted not to change this, but we can remove that check later if we don't think it impacts valid command executions.